src/store/modules/ModDownloadModule.ts

```diff
@@ -0,0 +1,112 @@
+import {ActionTree, GetterTree} from 'vuex';
+
+import {ImmutableProfile} from '../../model/Profile';
+import ThunderstoreCombo from '../../model/ThunderstoreCombo';
+import ThunderstoreDownloaderProvider from '../../providers/ror2/downloading/ThunderstoreDownloaderProvider';
+import {State as RootState} from '../index';
+import R2Error from "../../model/errors/R2Error";
+
+// TODO: Replace the "enum" in '../../model/enums/StatusEnum' with a proper enum like this one.
+enum StatusEnum {
+    FAILURE = 0,
+    PENDING = 1,
+    SUCCESS = 2,
+}
+
+interface ProgressItem {
+    modName: string;
+    installProgress: number;
+    downloadProgress: number;
+    status: number; // StatusEnum
+    error: R2Error | null;
+}
+
+interface State {
+    downloads: ProgressItem[];
+}
+
+/**
+ * State for Mod Downloading/Installation.
+ */
+export default {
```

Comment:
Nitpick: use named exports whenever possible. Default exports can lead to errors when imported carelessly.